### PR TITLE
New storage for memrem and list keys

### PR DESCRIPTION
**Pull request by nwest**

### CODE REVIEW

File: scripts/memrem.coffee [hunk 1]
```diff
@@ -10,30 +10,55 @@
...
+  key = msg.match[2]
+  if key is "list"
+    return
+
   value = msg.match[3]
-  key = memremId keyword
-  msg.robot.brain.set key, value
-  msg.send "saved #{value} as #{keyword}"
+  brain = msg.robot.brain
+  storage = brain.get memremBrain
```
Comment (macklinu) [inline scripts/memrem.coffee]: I think

```coffee
storage = brain.get memremBrain

if !storage
  storage = {}
```

could be transformed into

```coffee
storage = brain.get memremBrain ? {}
```

via http://coffeescript.org/#operators (specifically the existential operator)

Comment (nwest) [inline scripts/memrem.coffee]: thanks!

File: scripts/memrem.coffee [hunk 2]
```diff
@@ -10,30 +10,55 @@
...
+memremBrain = "memrem"
 
 onGet = (msg) ->
-  keyword = msg.match[2]
-  key = memremId keyword
-  value = msg.robot.brain.get key
-  if value
+  key = msg.match[2]
+
+  if key is "list"
```
Comment (macklinu) [inline scripts/memrem.coffee]: Here is the current set regex:

```
/(mem|get) (?!safesearch)([a-zA-Z0-9_-]*)$/i
```

Could we just ignore list in the regex and remove the `if key is "list"` check in both set and get commands by adding `list` to the regex, like so?

```
/(mem|get) (?!safesearch|list)([a-zA-Z0-9_-]*)$/i
```

**Comment by macklinu** [comment]
(yey) thanks for the rebase! LGTM

**Comment by macklinu** [comment]
@jsibbold @heymattlauer @whinchman @stkent NW added the list command + updated the regex to support hyphens and numbers, but it breaks backwards compatibility with all previously set memrem keys. This means merging the PR would break essentially start from scratch.

I like these changes, but as a casual user of memremgetset, I don't have a say in how (non)heartbroken more avid users would be if these changes were made.

Please weigh in on this PR - thanks!

**Comment by whinchman** [comment]
@macklinu, I'm okay with it. Sometimes, starting fresh leads to new inspirations. ¯\_(   ͠° ͟ ͟ʖ ͡° )_/¯

**Comment by nwest** [comment]
Also closes #157

**Comment by macklinu** [comment]
Merging this later today unless there are other objections.

**Comment by nwest** [comment]
#yolo

**Pull request merged**